src/Tensor.idr

```diff
@@ -689,128 +681,128 @@ export
 (-) : Primitive.Neg dtype => Tensor shape dtype -> Tensor shape dtype -> Tensor shape dtype
 (-) = binaryOp prim__sub
 
-infixl 9 *#, /#
+namespace Elementwise
+  ||| Element-wise multiplication. For example, `const [2, 3] * const [4, 5]` is equivalent to
+  ||| `const [8, 15]`.
+  export
+  (*) : Primitive.Num dtype
+        => Tensor (d :: ds) dtype -> Tensor (d :: ds) dtype -> Tensor (d :: ds) dtype
+  (*) = binaryOp prim__mul
 
-||| Element-wise multiplication. For example, `const [2, 3] *# const [4, 5]` is equivalent to
+||| Element-wise multiplication. For example, `const [2, 3] * const [4, 5]` is equivalent to
 ||| `const [8, 15]`.
 export
-(*#) : Primitive.Num dtype => Tensor shape dtype -> Tensor shape dtype -> Tensor shape dtype
-(*#) = binaryOp prim__mul
-
-||| Multiplication by a constant. For example, `const 2 * const [3, 5]` is equivalent to
-||| `const [6, 10]`.
-export
-(*) : Primitive dtype => Primitive.Num dtype =>
-      Tensor [] dtype -> Tensor shape dtype -> Tensor shape dtype
+(*) : (Primitive dtype, Primitive.Num dtype)
+      => Tensor [] dtype -> Tensor shape dtype -> Tensor shape dtype
 l * r with (r)
-  _ | MkTensor {shape} _ = broadcast {prf=scalarToAnyOk shape} l *# r
+  _ | (MkTensor {shape} _) = binaryOp prim__mul (broadcast {prf=scalarToAnyOk shape} l) r
 
 namespace Semigroup
   export
   [Prod] Primitive.Num dtype => Semigroup (Tensor shape dtype) where
-    (<+>) = (*#)
+    (<+>) = binaryOp prim__mul
```

Comment:
does this show that it's not very practical splitting multiplication between two APIs?